k4FWCore/src/PodioDataSvc.cpp

```diff
@@ -109,10 +126,16 @@ StatusCode PodioDataSvc::i_setRoot(std::string root_path, DataObject* pRootObj)
 }
 
 void PodioDataSvc::endOfRead() {
+  m_eventNum++;
+
+  if (!m_bounds_check_needed) {
+    return;
+  }
+
   StatusCode sc;
```

Comment:
This can be inside the loop, there is also a comment about this being unused that I don't think is relevant anymore

Reply:
I could put this inside of the if yes, I just did not want to make more changes than necessary. 
The comment is still valid I think. The variable is only there to hold the `StatusCode`(s) returned by
```
      sc = service("ApplicationMgr", eventProcessor);
      sc = eventProcessor->stopRun()
```
I guess we should check them and act accordingly, but I gladly leave that change to someone else.